### PR TITLE
feat: Batch processing for an anchor loop

### PR DESCRIPTION
**Pull request by ukstv**
batch size = concurrency

### CODE REVIEW

File: packages/core/src/anchor/processing-loop.ts [hunk 1]
```diff
@@ -107,12 +107,13 @@ export class ProcessingLoop<T> {
...
-          const value = next.value
-          if (!value) {
+          const values = next.value
+          if (!values) {
             this.#logger.verbose(`No value received in ProcessingLoop, skipping this iteration`)
             continue
           }
-          await Promise.race([this.handleValue(value), rejectOnAbortSignal])
+          const handlersP = values.map((value) => this.handleValue(value))
+          await Promise.race(handlersP.concat(rejectOnAbortSignal))
```
Comment (dbcfd) [inline packages/core/src/anchor/processing-loop.ts:116]: Shouldn't this be 

`await Promise.race(Promise.all(handlersP), rejectOnAbortSignal)`

We want all the handlers to finish, but we want to race between handlers finishing and being aborted.

**Review by dbcfd — CHANGES_REQUESTED** [review]

File: packages/core/src/anchor/processing-loop.ts (hunk 1 above)
Comment (ukstv) [reply packages/core/src/anchor/processing-loop.ts:116]: yes!

Comment (ukstv) [reply packages/core/src/anchor/processing-loop.ts]: there now.

**Review by dbcfd — APPROVED** [review]

**Comment by stbrody** [comment]
This works, though I have two (admittedly both on the relatively minor side) issues with the approach. 

One concern is that combining the batch size used for fetching requests from the store and the concurrency limit of processing those requests into a single knob limits us.  I think it makes sense to fetch a large number of requests from the store at once (like the 1000 value we previously had here), but we probably want to limit the concurrency of how many requests get processed at once quite a bit more than that, since we don't want to overwhelm the CAS or use too much memory on the Ceramic node holding all the CAR file responses from that many requests in memory at once.

The other possible improvement I see over this approach is that rather than doing things in batches where every single request from the batch must be fully processed before we start any requests in a new batch, we can instead specify the number of requests we want processing concurrently and then keep a steady supply of requests always in processing up to that limit.  That can keep us busier, processing things faster, and smoothing out the load.

I took a crack at implementing an alternative approach that addresses these two issues here: https://github.com/ceramicnetwork/js-ceramic/pull/3153, PTAL.

**Comment by ukstv** [comment]
Okay.

**Pull request closed without merging**